### PR TITLE
KOGITO-1314+KOGITO-1353: Use kogito-bom in quickstarts, fix Custom Lifecycle

### PR DESCRIPTION
**Pull request by evacchi**
As a (temporary?) measure before we are part of Quarkus Universe BOM, let's use our own BOM with the correct versions. For now we'll need to point to a fixed Quarkus version, too.

### CODE REVIEW

File: kogito-business-rules-quarkus/pom.xml [hunk 1]
```diff
@@ -18,14 +18,15 @@
 
 
   <properties>
+    <kogito.version>8.0.0-SNAPSHOT</kogito.version>
```
Comment (radtriste) [inline kogito-business-rules-quarkus/pom.xml]: already defined into parent pom. No need to rewrite it here or ?

**Review by radtriste — APPROVED** [review]
lgtm

one small thing

**Review by radtriste — APPROVED** [review]
seems ok to me.
usertasks-custom-lifecycle-* tests are failing, but not sure it is related to this change will need further investigation

**Comment by radtriste** [comment]
@evacchi @mariofusco should we merge that one ?

**Comment by evacchi** [comment]
@radtriste even if, I don't have merge rights on this repo

**Comment by evacchi** [comment]
I am disabling custom lifecycle quickstarts as they were already broken with 0.7.0 but seeking help from @mswiderski ( <3 ) to understand what's going wrong there. I have fixed the auth problem:

```java
if (!targetPhase.id().equals(Active.ID) && !targetPhase.id().equals(Abort.ID) && !humanTaskWorkItem.enforce(transition.policies().toArray(new Policy[transition.policies().size()]))) {
     throw new NotAuthorizedException("User is not authorized to access task instance with id " + humanTaskWorkItem.getId());
} 
```

but now instead of 
```
Caused by: org.kie.kogito.process.workitem.NotAuthorizedException: User is not authorized to access task instance with id 771ac2d7-858e-4eb0-97f1-f71b2852bd0b
```
we are getting
```
org.kie.kogito.process.workitem.InvalidTransitionException: Cannot transition from active to complete
	at org.acme.travels.ApprovalsProcessTest.testApprovalProcess(ApprovalsProcessTest.java:61)
```
this was already broken in 0.7.0 apparently -- I tested with #3

**Comment by mswiderski** [comment]
maybe this commit will help https://github.com/mswiderski/kogito-quickstarts/commit/9147069f9e9f0908d657570e90540c231d351db7

**Comment by evacchi** [comment]
with that commit applied I am getting 2 errors :(

```
[INFO] Running org.acme.travels.ApprovalsProcessTest
2020-03-04 16:42:16,318 INFO  [org.kie.kog.cod.pro.ProcessCodegen] (build-4) Generating REST Resources.
2020-03-04 16:42:20,173 INFO  [io.quarkus] (main) Quarkus 1.3.0.Alpha2 started in 6.021s. Listening on: http://0.0.0.0:8081
2020-03-04 16:42:20,174 INFO  [io.quarkus] (main) Profile test activated. 
2020-03-04 16:42:20,174 INFO  [io.quarkus] (main) Installed features: [cdi, kogito, resteasy, resteasy-jackson, smallrye-openapi, swagger-ui]
[ERROR] Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 9.718 s <<< FAILURE! - in org.acme.travels.ApprovalsProcessTest
[ERROR] testApprovalProcess  Time elapsed: 0.391 s  <<< ERROR!
org.kie.kogito.process.workitem.InvalidLifeCyclePhaseException: start
	at org.acme.travels.ApprovalsProcessTest.testApprovalProcess(ApprovalsProcessTest.java:61)

[ERROR] testApprovalProcessViaPhases  Time elapsed: 0.01 s  <<< ERROR!
java.lang.RuntimeException: org.opentest4j.AssertionFailedError: Expected org.kie.kogito.process.workitem.InvalidTransitionException to be thrown, but nothing was thrown.
	at org.acme.travels.ApprovalsProcessTest.testApprovalProcessViaPhases(ApprovalsProcessTest.java:112)
```

**Comment by mswiderski** [comment]
do you see the same errors in spring boot version? I just run tests on that and all pass. I haven't migrated to new kogito extension ...

**Comment by evacchi** [comment]
springboot works -- that's... weird. BTW the new extension is in-tree now, so if you have time to pull my branch of this repo it's pretty easy to see (it will pull the runtimes snapshot). I really don't understand what's going on, though

**Comment by evacchi** [comment]
ok my bad I had inadvertently committed a change to the quarkus version 😅 thanks @mswiderski

**Review by radtriste — APPROVED** [review]

**Pull request merged**